File: Android/EasyBudget/app/src/main/java/com/ajapplications/budgeteerbuddy/view/welcome/Onboarding2Fragment.java
/*
 *   Copyright 2015 Benoit LETONDOR
 *
 *   Licensed under the Apache License, Version 2.0 (the "License");
 *   you may not use this file except in compliance with the License.
 *   You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *   Unless required by applicable law or agreed to in writing, software
 *   distributed under the License is distributed on an "AS IS" BASIS,
 *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *   See the License for the specific language governing permissions and
 *   limitations under the License.
 */

package com.ajapplications.budgeteerbuddy.view.welcome;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentTransaction;
import android.support.v4.content.LocalBroadcastManager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import com.ajapplications.budgeteerbuddy.helper.CurrencyHelper;
import com.ajapplications.budgeteerbuddy.view.selectcurrency.SelectCurrencyFragment;
import com.ajapplications.budgeteerbuddy.R;

import java.util.Currency;

/**
 * Onboarding step 2 fragment
 *
 * @author Benoit LETONDOR
 */
public class Onboarding2Fragment extends OnboardingFragment
{
    private Currency selectedCurrency;
    private Button nextButton;

    private BroadcastReceiver receiver;

// ------------------------------------->

    /**
     * Required empty public constructor
     */
    public Onboarding2Fragment()
    {

    }

// ------------------------------------->

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
    {
        // Inflate the layout for this fragment
        View v = inflater.inflate(R.layout.fragment_onboarding2, container, false);

        nextButton = (Button) v.findViewById(R.id.onboarding_screen2_next_button);
        nextButton.setOnClickListener(new View.OnClickListener()
        {
            @Override
            public void onClick(View v)
            {
                next();
            }
        });

        selectedCurrency = CurrencyHelper.getUserCurrency(v.getContext());
        setNextButtonText();

        Fragment selectCurrencyFragment = new SelectCurrencyFragment();
        FragmentTransaction transaction = getChildFragmentManager().beginTransaction();
        transaction.add(R.id.expense_select_container, selectCurrencyFragment).commit();

        IntentFilter filter = new IntentFilter(SelectCurrencyFragment.CURRENCY_SELECTED_INTENT);
        receiver = new BroadcastReceiver()
        {
            @Override
            public void onReceive(Context context, Intent intent)
            {
                selectedCurrency = Currency.getInstance(intent.getStringExtra(SelectCurrencyFragment.CURRENCY_ISO_EXTRA));
                setNextButtonText();
            }
        };

        LocalBroadcastManager.getInstance(v.getContext()).registerReceiver(receiver, filter);

        return v;
    }

    @Override
    public void onDestroyView()
    {
        LocalBroadcastManager.getInstance(getActivity()).unregisterReceiver(receiver);

        super.onDestroyView();
    }

    @Override
    public int getStatusBarColor()
    {
        return R.color.secondary_dark;
    }

    /**
     * Set the next button text according to the selected currency
     */
    private void setNextButtonText()
    {
        nextButton.setText(getResources().getString(R.string.onboarding_screen_2_cta, selectedCurrency.getSymbol()));
    }
}

File: Android/EasyBudget/app/src/main/java/com/benoitletondor/easybudgetapp/view/welcome/Onboarding2Fragment.kt
/*
 *   Copyright 2024 Benoit LETONDOR
 *
 *   Licensed under the Apache License, Version 2.0 (the "License");
 *   you may not use this file except in compliance with the License.
 *   You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *   Unless required by applicable law or agreed to in writing, software
 *   distributed under the License is distributed on an "AS IS" BASIS,
 *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *   See the License for the specific language governing permissions and
 *   limitations under the License.
 */

package com.benoitletondor.easybudgetapp.view.welcome

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.localbroadcastmanager.content.LocalBroadcastManager
import com.benoitletondor.easybudgetapp.R
import com.benoitletondor.easybudgetapp.databinding.FragmentOnboarding2Binding
import com.benoitletondor.easybudgetapp.parameters.Parameters
import com.benoitletondor.easybudgetapp.helper.getUserCurrency
import com.benoitletondor.easybudgetapp.view.selectcurrency.SelectCurrencyFragment
import dagger.hilt.android.AndroidEntryPoint
import java.util.*
import javax.inject.Inject

/**
 * Onboarding step 2 fragment
 *
 * @author Benoit LETONDOR
 */
@AndroidEntryPoint
class Onboarding2Fragment : OnboardingFragment<FragmentOnboarding2Binding>() {
    private lateinit var selectedCurrency: Currency
    private lateinit var receiver: BroadcastReceiver

    @Inject lateinit var parameters: Parameters

    override val statusBarColor: Int
        get() = R.color.secondary_dark

// ------------------------------------->

    override fun onCreateBinding(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): FragmentOnboarding2Binding = FragmentOnboarding2Binding.inflate(inflater, container, false)

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        selectedCurrency = parameters.getUserCurrency()
        setNextButtonText()

        val selectCurrencyFragment = SelectCurrencyFragment()
        val transaction = childFragmentManager.beginTransaction()
        transaction.add(R.id.expense_select_container, selectCurrencyFragment).commit()

        val filter = IntentFilter(SelectCurrencyFragment.CURRENCY_SELECTED_INTENT)
        receiver = object : BroadcastReceiver() {
            override fun onReceive(context: Context, intent: Intent) {
                selectedCurrency = Currency.getInstance(intent.getStringExtra(SelectCurrencyFragment.CURRENCY_ISO_EXTRA))
                setNextButtonText()
            }
        }

        LocalBroadcastManager.getInstance(view.context).registerReceiver(receiver, filter)

        binding?.onboardingScreen2NextButton?.setOnClickListener {
            next()
        }
    }

    override fun onDestroyView() {
        LocalBroadcastManager.getInstance(requireContext()).unregisterReceiver(receiver)

        super.onDestroyView()
    }

    /**
     * Set the next button text according to the selected currency
     */
    private fun setNextButtonText() {
        binding?.onboardingScreen2NextButton?.text = resources.getString(R.string.onboarding_screen_2_cta, selectedCurrency.symbol)
    }
}
